### PR TITLE
Add Option

### PR DESCRIPTION
**Pull request by onmyway133**
- Add `Option` to specify options when fetching image for `imageView`.
- The old `Configuration` is for global thing, like tracking and default storage.
- With `Option`, users have the chance to specify `ImageProcessing` and `ImageDisplaying` object to handle
- There's default `SimpleImageDisplayer` that allows animation options to be passed
- The `ImageDrawer` has been removed, ping @zenangst , in favor of `TintImageProcessor`
- Add tests for `SimpleImageDisplayer` and `ImageView+Imaginary`

### CODE REVIEW

**Comment by onmyway133** [comment]
I really like this PR, each has single responsibility, with `Option` and `Configuration`. The `Option` also allows injecting your own `ImageFetcher`. And there are tests 🎉  Let me know what you think @hyperoslo/ios ?

File: ImaginaryTests-iOS/ImageView+ImaginaryTests.swift [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import XCTest
+import Cache
+@testable import Imaginary
+
+private final class ImageView_ImaginaryTests: XCTestCase {
```
Comment (vadymmarkov) [inline ImaginaryTests-iOS/ImageView+ImaginaryTests.swift:5]: I see you like underscores.

File: ImaginaryTests-iOS/TestHelper.swift [hunk 2]
```diff
@@ -14,3 +15,13 @@ class TestHelper {
     return image!
   }
 }
+
+class MockDownloader: ImageDownloader {
```
Comment (vadymmarkov) [inline ImaginaryTests-iOS/TestHelper.swift:19]: private?

File: Sources/Mac/Decompressor.swift [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import Cocoa
 
-struct Decompressor {
-  static func decompress(_ data: Data, scale: CGFloat = 1) -> NSImage? {
+class Decompressor {
```
Comment (vadymmarkov) [inline Sources/Mac/Decompressor.swift]: `final class`?

File: Sources/Shared/Library/Option.swift [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import Foundation
+import Cache
+
+/// Option applied when fetching image
+public struct Option {
```
Comment (vadymmarkov) [inline Sources/Shared/Library/Option.swift:5]: Not sure it's the best name.

File: Sources/iOS/TintImageProcessor.swift [hunk 5]
```diff
@@ -0,0 +1,57 @@
+import UIKit
+
+/// Apply tint color to image
+public class TintImageProcessor: ImageProcessing {
+  let tintColor: UIColor
```
Comment (vadymmarkov) [inline Sources/iOS/TintImageProcessor.swift]: `private`?

File: Sources/iOS/Decompressor.swift [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import UIKit
 
-struct Decompressor {
-  static func decompress(_ data: Data, scale: CGFloat = 1) -> Image? {
+class Decompressor {
```
Comment (vadymmarkov) [inline Sources/iOS/Decompressor.swift]: `final`?

**Review by vadymmarkov — APPROVED** [review]

File: ImaginaryTests-iOS/TestHelper.swift (hunk 2 above)
Comment (onmyway133) [reply ImaginaryTests-iOS/TestHelper.swift:19]: @vadymmarkov It is used by many tests, so let's keep it `internal`

File: ImaginaryTests-iOS/ImageView+ImaginaryTests.swift (hunk 1 above)
Comment (onmyway133) [reply ImaginaryTests-iOS/ImageView+ImaginaryTests.swift:5]: Did you know I'm also a fan of http://underscorejs.org/ 😉

File: Sources/iOS/TintImageProcessor.swift (hunk 5 above)
Comment (onmyway133) [reply Sources/iOS/TintImageProcessor.swift]: 👍

File: Sources/Mac/Decompressor.swift (hunk 3 above)
Comment (onmyway133) [reply Sources/Mac/Decompressor.swift]: 👍

File: Sources/iOS/Decompressor.swift (hunk 6 above)
Comment (onmyway133) [reply Sources/iOS/Decompressor.swift]: 👍

**Pull request merged**